azure/converters/vmss.go

```diff
@@ -149,8 +158,52 @@ func SDKToVMSSVM(sdkInstance compute.VirtualMachineScaleSetVM) *azure.VMSSVM {
 
 // SDKImageToImage converts a SDK image reference to infrav1.Image.
 func SDKImageToImage(sdkImageRef *compute.ImageReference, isThirdPartyImage bool) infrav1.Image {
+	if sdkImageRef.ID != nil {
```

Comment:
Is this referring to a compute gallery image? If so, a comment here could help distinguish the `ID` from community and shared images.